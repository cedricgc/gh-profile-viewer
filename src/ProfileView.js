import React, { Component } from 'react';

import { gql } from 'apollo-boost';
import { Query } from 'react-apollo';

class ProfileView extends Component {
  render() {
    if (this.props.username === '') {
      return null;
    }

    const GET_USER = gql`
      query {
        user(login: "${this.props.username}") {
          id
          name
          url
        }
      }
    `;

    return (
      <div id="gh-profile" data-test-id="profile-container">
        <Query query={GET_USER}>
          {({ loading, error, data }) => {
            if (loading) {
              return (
                <div>
                  <span>Loading...</span>
                </div>
              );
            }
            if (error) {
              return (
                <div>
                  <span>
                    Error: Unable to load user {`${this.props.username}`}
                  </span>
                </div>
              );
            }

            return (
              <div>
                <h2>Github Profile: {this.props.username}</h2>
                <div>
                  <span>Github ID: {data.user.id}</span>
                </div>
              </div>
            );
          }}
        </Query>
      </div>
    );
  }
}

export default ProfileView;
