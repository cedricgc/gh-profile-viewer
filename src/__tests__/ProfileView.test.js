import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import { ProfileView } from '../ProfileView';

test('component matches snapshot', () => {
  const render = ReactTestRenderer.create(
    <ProfileView
      username={'user'}
      data={{ user: { id: 'IDXXXX' }, loading: false, error: false }}
    />
  );

  const tree = render.toJSON();
  expect(tree).toMatchSnapshot();
});
