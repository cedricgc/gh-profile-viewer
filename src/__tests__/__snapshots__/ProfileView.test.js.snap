// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component matches snapshot 1`] = `
<div
  data-test-id="profile-container"
  id="gh-profile"
>
  <div>
    <h2>
      Github Profile: 
      user
    </h2>
    <div>
      <span>
        Github ID: 
        IDXXXX
      </span>
    </div>
  </div>
</div>
`;
